src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -189,6 +189,444 @@ public static class ILoggerExtensions
+            this ILogger logger,
+            string serviceBusNamespace,
+            string queueName,
+            string operationName,
+            bool isSuccessful,
+            TimeSpan duration,
+            DateTimeOffset startTime,
+            Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger), "Requires an logger instance to track telemetry");
+            Guard.NotNullOrWhitespace(serviceBusNamespace, nameof(serviceBusNamespace), "Requires an Azure Service Bus namespace to track the queue request");
+            Guard.NotNullOrWhitespace(queueName, nameof(queueName), "Requires an Azure Service Bus queue name to track the queue request");
+            Guard.NotLessThan(duration, TimeSpan.Zero, nameof(duration), "Requires a positive time duration of the Azure Service Bus queue request operation");
+
+            LogServiceBusRequest(logger, serviceBusNamespace, queueName, operationName, isSuccessful, duration, startTime, ServiceBusEntityType.Queue, context);
+        }
+
+        /// <summary>
+        ///     Logs an Azure Service Bus request.
+        /// </summary>
+        /// <param name="logger">The logger instance to track the telemetry.</param>
+        /// <param name="serviceBusNamespace">The namespace where the Azure Service Bus queue is registered. (Will be combined with <paramref name="serviceBusNamespaceSuffix"/>).</param>
+        /// <param name="serviceBusNamespaceSuffix">The namespace suffix (i.e. '.servicebus.windows.net') where the Azure Service Bus queue is registered (Will be combined with <paramref name="serviceBusNamespace"/>).</param>
+        /// <param name="entityName">The name of the Azure Service Bus entity.</param>
+        /// <param name="operationName">The name of the operation of the request.</param>
+        /// <param name="isSuccessful">The indication whether or not the Azure Service Bus request was successfully processed.</param>
+        /// <param name="measurement">The instance to measure the latency duration of the Azure Service Bus queue request.</param>
+        /// <param name="entityType">The type of the Azure Service Bus entity.</param>
+        /// <param name="context">The telemetry context that provides more insights on the Azure Service Bus request.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="logger"/> or the <paramref name="measurement"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="serviceBusNamespace"/> or <paramref name="entityName"/> is blank.</exception>
+        public static void LogServiceBusRequestWithSuffix(
+            this ILogger logger,
+            string serviceBusNamespace,
+            string serviceBusNamespaceSuffix,
+            string entityName,
+            string operationName,
+            bool isSuccessful,
+            DependencyMeasurement measurement,
```

Comment:
Should we create a `RequestMeasurement` instead ?  Or can we make the `DependencyMeasurement` type more generic so that it can be renamed to `Measurement` ?

Reply:
Created an issue for this #268